lib/src/graph/package_graph.dart

```diff
@@ -168,9 +168,10 @@ class PackageGraph {
       return new Future.error(error, _lastUnexpectedErrorTrace);
     }
 
-    // If the last build completed with an error, complete the future with it.
-    if (!_lastResult.succeeded) {
-      return new Future.error(BarbackException.aggregate(_lastResult.errors));
+    // If any build completed with an error, complete the future with it.
+    var result = new BuildResult.aggregate(_results);
```

Comment:
In pub's usage, this may only get hit once, but barback is a general library that others may use as well and they may call `getAllAssets()` multiple times. If they do, I don't think previous failures should re-appear in later calls.